Description: Fix floating point precision issue in tests
 Introduce precision tolerance to fix floating point
 error in tests in i386.
Author: Yogeswaran Umasankar <yogu@debian.org>
Forwarded: not-needed
Last-Update: 2024-12-13

---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/src/distribution/gumbel.rs
+++ b/src/distribution/gumbel.rs
@@ -389,11 +389,12 @@ mod tests {
     #[test]
     fn test_mean() {
         let mean = |x: Gumbel| x.mean().unwrap();
-        test_exact(0.0, 2.0, 1.1544313298030658, mean);
-        test_exact(0.1, 4.0, 2.4088626596061316, mean);
-        test_exact(1.0, 10.0, 6.772156649015328, mean);
-        test_exact(10.0, 11.0, 16.34937231391686, mean);
-        test_exact(10.0, f64::INFINITY, f64::INFINITY, mean);
+        let acc = 1e-12;
+        test_absolute(0.0, 2.0, 1.1544313298030658, acc, mean);
+        test_absolute(0.1, 4.0, 2.4088626596061316, acc, mean);
+        test_absolute(1.0, 10.0, 6.772156649015328, acc, mean);
+        test_absolute(10.0, 11.0, 16.34937231391686, acc, mean);
+        test_absolute(10.0, f64::INFINITY, f64::INFINITY, acc, mean);
     }
 
     #[test]
@@ -427,11 +428,12 @@ mod tests {
     #[test]
     fn test_median() {
         let median = |x: Gumbel| x.median();
-        test_exact(0.0, 2.0, 0.7330258411633287, median);
-        test_exact(0.1, 4.0, 1.5660516823266574, median);
-        test_exact(1.0, 10.0, 4.665129205816644, median);
-        test_exact(10.0, 11.0, 14.031642126398307, median);
-        test_exact(10.0, f64::INFINITY, f64::INFINITY, median);
+        let acc = 1e-12;
+        test_absolute(0.0, 2.0, 0.7330258411633287, acc, median);
+        test_absolute(0.1, 4.0, 1.5660516823266574, acc, median);
+        test_absolute(1.0, 10.0, 4.665129205816644, acc, median);
+        test_absolute(10.0, 11.0, 14.031642126398307, acc, median);
+        test_absolute(10.0, f64::INFINITY, f64::INFINITY, acc, median);
     }
 
     #[test]
